core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -606,7 +608,11 @@ private Query parse(AliasMetaData alias, QueryShardContext shardContext) {
         try {
             byte[] filterSource = alias.filter().uncompressed();
             try (XContentParser parser = XContentFactory.xContent(filterSource).createParser(filterSource)) {
-                ParsedQuery parsedFilter = shardContext.toFilter(shardContext.newParseContext(parser).parseInnerQueryBuilder());
+                Optional<QueryBuilder> innerQueryBuilder = shardContext.newParseContext(parser).parseInnerQueryBuilder();
+                ParsedQuery parsedFilter = null;
+                if (innerQueryBuilder.isPresent()) {
+                    parsedFilter = shardContext.toFilter(innerQueryBuilder.get());
+                }
```

Comment:
I think we should consider null a bug here. I don't see good reasons why doToQuery should return null. Best fix would be to fix that at the source instead of having null check in QueryShardContent#toQuery . What does null mean in CommonTermsQueryBuilder? it should be responsibility of each query to construct a proper non null lucene query I think.